core/commands/ipns.go

```diff
@@ -80,7 +86,7 @@ Resolve the value of another name:
 			depth = namesys.DefaultDepthLimit
 		}
 
-		resolver := namesys.NewRoutingResolver(n.Routing)
+		resolver := namesys.NewRoutingResolver(router)
```

Comment:
i suspect this will break if resolvers start doing other things to the ipfs node. let's probably rethink what makes up an ipfs node and how the commands run pieces of code. 

I would like to move all of the _command logic/behavior_ out of the cmdslib specific part, into -- say-- `core`, and define a proper interface. maybe post 0.4.0